lib/racecar/runner.rb

```diff
@@ -2,12 +2,17 @@
 
 module Racecar
   class Runner
-    attr_reader :processor, :config, :logger
+    attr_reader :processor, :config, :logger, :consumer
 
     def initialize(processor, config:, logger:)
       @processor, @config, @logger = processor, config, logger
     end
 
+    def stop
+      processor.stop if processor.respond_to?(:stop)
```

Comment:
Could we call this callback e.g. `teardown`? I think that better communicates that it's a callback that allows tearing down resources, not a method that actually stops anything.